### PR TITLE
Make schemas mandatory and use schema API from ROM core

### PR DESCRIPTION
**Pull request by AMHOL**

### CODE REVIEW

**Comment by AMHOL** [comment]
@solnic let me know what you think when you get a chance

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 [rom]:  https://github.com/rom-rb/rom
 
 
-# ROM-http [![Gitter chat](https://badges.gitter.im/rom-rb/chat.svg)][gitter]
+# ROM-HTTP [![Gitter chat](https://badges.gitter.im/rom-rb/chat.svg)][gitter]
```
Comment (solnic) [inline README.md]: actually, we have lowcased name in other adapter repos :)

File: README.md [hunk 2]
```diff
@@ -69,23 +70,18 @@ end
 
 class ResponseHandler
   def call(response, dataset)
-    Array([JSON.parse(response.body)]).flatten
+    Array([JSON.parse(response.body, symbolize_names: true)]).flatten
```
Comment (solnic) [inline README.md]: OMG I had no idea there's a `symbolize_names` option!

Comment (AMHOL) [inline README.md]: Yeah found out a while back, makes things a lot easier than trying to handle it yourself lol

**Pull request merged**